src/spartan_app.erl

```diff
@@ -103,12 +113,9 @@ load_json_config(FileBin) ->
     lists:foreach(fun process_config_tuple/1, ConfigTuples).
 
 process_config_tuple({<<"upstream_resolvers">>, UpstreamResolvers}) ->
-    UpstreamResolverIPs = lists:map(fun parse_ipv4_address/1, UpstreamResolvers),
-    ConfigValue = [{UpstreamResolverIP, 53} || UpstreamResolverIP <- UpstreamResolverIPs],
-    application:set_env(?APP, upstream_resolvers, ConfigValue);
+    UpstreamIpsAndPorts = [parse_ipv4_address_with_port(Resolver, 53) || Resolver <- UpstreamResolvers],
+    application:set_env(?APP, upstream_resolvers, UpstreamIpsAndPorts);
 process_config_tuple({Key, Value}) when is_binary(Value) ->
     application:set_env(?APP, binary_to_atom(Key, utf8), binary_to_list(Value));
 process_config_tuple({Key, Value}) ->
     application:set_env(?APP, binary_to_atom(Key, utf8), Value).
```

Comment:
Mind adding some eunit tests? I think you can copy and paste the ones from pr #12 for the most part.